### DISCUSSION TITLE
query context

### DISCUSSION THREAD

**Discussion by correttojs** [TanStack/query#2286] [Ideas]
Hi,

I'm writing a [gql client](https://github.com/correttojs/react-query-gql) based on fetch and react-query.

I was considering an example with multiple gql client instances, each of them linked to a different endpoint. 

It would be nice to expose a `context` to share some custom data between the react-query `QueryClient ` and the `queryFn` (so extending the QueryFunctionContext) and `mutationFn`. This way, in the above use case, I could define a default endpoint in the scope of the default react query client, and customize it in case I want to access a different endpoint.

The `queryFn` already works in a similar way, as it can define a default implementation and override it on each `useQuery`, though it doesn't seem to be flexible enough for my use case

**Comment by TkDodo** [comment]
could you elaborate why the default query function is not flexible enough? I don't fully understand the use case yet
